api-service/webapp-demo/src/app/base_job.component.ts

```diff
@@ -133,9 +135,22 @@ export class BaseJobComponent implements OnInit {
     this.questionId = params.get('questionId');
   }
 
-  public sendScoreToApi() {
-    console.log('test click');
+  updateUrl(): void {
+    if (!this.customClientJobKey) {
+      return;
+    }
+    const urlParams: JobAndQuestionUrlParams = {
+      customClientJobKey: this.customClientJobKey,
+    };
+
+    if (this.questionId) {
+      urlParams.questionId = this.questionId;
+    }
 
+    this.router.navigate([this.routerPath, urlParams]);
+  }
```

Comment:
Just curious what the reasoning was for moving around the `updateUrl` function here. Was it not always running?

Reply:
Now I'm trying to remember why I did this...I remember I was having some trouble with url parameters getting updated in the test / not getting set, and this was part of my troubleshooting. But now if I remove the updateUrl() call from ngOnInit, the tests still pass, so it's possible that was a red herring--the problem was actually fixed with ActivatedRouteStub. I'll remove the call from ngOnInit.